app/backend/db/crud/date_datas.py

```diff
@@ -83,27 +93,42 @@ def delete_users_dates_data(
 
         for date_data in dates_data_to_remove:
             db.delete(date_data)
+    
+    db.commit()
 
 def put_data_in_user(
     db: Session, 
-    body=schemas.PutDateDataBody
+    start_date: date,
+    end_date: date = None,
+    state: schemas.AnswerStateTypes = None,
+    reason: str = None,
+    reported_by_id: int = None,
+    reported_time: datetime = None
 ) -> schemas.RangeDatesResponse:
 
-    request = body.dict()
-    if body.reason is not None:
-        request["reason"] = get_reason_by_name(db, body.reason)
+
+    if reason is not None:
+        reason = get_reason_by_name(db, reason)
 
     dates_data = get_dates_data(
             db=db, 
-            user_id=request.pop("user_id"),
-            start_date=request.pop("start_date"),
-            end_date=request.pop("end_date")
+            user_id=user_id,
+            start_date=start_date,
+            end_date=end_date
         )
 
     for datedata in dates_data:
-        for key, value in body.dict().items():
-            if value is not None:
-                setattr(datedata, key, value)
+        if state is not None:
+            datedata.state = state
+
+        if reason is not None:
+            datedata.reason = reason
+
+        if reported_by_id is not None:
+            datedata.reported_by_id = reported_by_id
+
+        if reported_time is not None:
+            datedata.reported_time = reported_time
```

Comment:
cant u do it just with .update function??
just do a list cleanups of all None

Reply:
i wanted but you told me not to..
![image](https://user-images.githubusercontent.com/31568827/71623328-62e60100-2be4-11ea-9511-ba392c636163.png)